rcl_lifecycle/src/default_state_machine.c

```diff
@@ -0,0 +1,295 @@
+// Copyright 2016 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <stdlib.h>
+#include <stdio.h>
+#include <string.h>
+
+#include <lifecycle_msgs/msg/state.h>
+#include <lifecycle_msgs/msg/transition.h>
+
+#include "rcl_lifecycle/transition_map.h"
+
+#include "default_state_machine.h"  // NOLINT
+#include "states.h"  // NOLINT
```

Comment:
Same comment about doing the relative include so that you do not accidentally get this header from somewhere on the include path.